### PR TITLE
Update versioninfo api to check platform tiers by new folder structure instead by .download_info

### PR DESCRIPTION
**Pull request by mcdurdin**

### CODE REVIEW

File: api/versioninfo.php [hunk 1]
```diff
@@ -9,6 +9,7 @@
   header('Cache-Control: max-age=0');
   
   $allowed_platforms = array('android', 'ios', 'mac', 'web', 'windows');
+  $tiers = array('alpha', 'beta', 'stable');
```
Comment (darcywong00) [inline api/versioninfo.php]: maybe $release_tiers instead of $tiers?

File: api/versioninfo.php [hunk 2]
```diff
@@ -34,7 +35,7 @@ function fail($s) {
   function version_filter($a) {
     global $platform;
     if($a == '.' || $a == '..') return false;
-    return is_dir("../$platform/$a");
+    return preg_match('/^\d+\.\d+(\.\d+)*$/', $a);
```
Comment (darcywong00) [inline api/versioninfo.php]: a comment about the regex might be helpful for maintenance

File: api/versioninfo.php [hunk 3]
```diff
@@ -49,49 +50,18 @@ function remove_utf8_bom($text) {
...
-          }
-        }
-        
-        break;
+    foreach($tiers as $tier) {
+      $dirs = scandir("../$platform/$tier");
+      $dirs = array_filter($dirs, 'version_filter');
+      if(count($dirs) > 0) {
+        usort($dirs, 'version_compare_backward');
+        $p[$tier] = $dirs[0];
```
Comment (darcywong00) [inline api/versioninfo.php:62]: assuming scandir is already giving you ascending versions, you could avoid `usort` with
```
$p[$tier] = array_pop($dirs);
```

**Review by darcywong00 — APPROVED** [review]
LGTM with minor comments

File: api/versioninfo.php (hunk 3 above)
Comment (mcdurdin) [reply api/versioninfo.php:62]: scandir gives alphabetical sort, which doesn't work with version strings (e.g. '9.5' < '10.0' but alphabetical sort doesn't do that)

Comment (darcywong00) [reply api/versioninfo.php:62]: gotcha.  How about scandir with SCANDIR_SORT_DESCENDING then?

**Pull request merged**